### PR TITLE
Dev/builtin session refresh

### PR DESCRIPTION
**Pull request by calvinleungyk**

### CODE REVIEW

File: omega_client/messaging/single_client_response_handler.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                logger.error(
+                    'authorization_grant error: ',
+                    extra={'message': authorization_grant.message})
+        self.on_authorization_grant(
+            authorization_grant=authorization_grant,
+            client_id=client_id,
+            sender_comp_id=sender_comp_id,
+            request_id=request_id)
+
+    def _send_authorization_refresh(self):
```
Comment (dev1-fund3) [inline omega_client/messaging/single_client_response_handler.py]: why is there a local call to _send_authorization_refresh instead just calling the method in request_sender directly? Seems redundant

File: omega_client/messaging/single_client_response_handler.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        :param sender_comp_id: (str) sender_comp_id of the response.
+        :param request_id: (int) request_id which requested this response
+        """
+        if authorization_grant and authorization_grant.success:
+            self._request_sender.set_access_token(
+                authorization_grant.access_token)
+            self._refresh_token = authorization_grant.refresh_token
+            self._token_expire_time = authorization_grant.expire_at
+            time_until_session_refresh = (self._token_expire_time -
+                                          dt.utcnow().timestamp() - 30.)
```
Comment (dev1-fund3) [inline omega_client/messaging/single_client_response_handler.py]: Make the 30. a REFRESH_BUFFER_TIME constant at the top.

File: omega_client/messaging/single_client_response_handler.py (hunk 1 above)
Comment (calvinleungyk) [reply omega_client/messaging/single_client_response_handler.py]: @dev1-fund3 It's for the delayed timer execution here:  
`Timer(time_until_session_refresh, self._send_authorization_refresh).start()`

File: omega_client/messaging/single_client_response_handler.py (hunk 2 above)
Comment (calvinleungyk) [reply omega_client/messaging/single_client_response_handler.py]: Done

**Pull request merged**